### PR TITLE
[69364] Feature: Repository responsibility of children nodes removal

### PR DESCRIPTION
**Pull request by lbajsarowicz**
PR depends on #111 

With this PR I removed the responsibility of removing children nodes from Controller, to Repository.

Controller's only responsibility should be navigating to the right resource / service to handle the request. Thus, there should be no business logic in Controller.

That is why I extracted this responsibility to Repository that responsibility is to maintain resources.

Also: replaced some weird "magic" behind the filtering with simple collection filter.

### CODE REVIEW

**Comment by snowremi** [comment]
Internal ticket DEV-69364

**Comment by lbajsarowicz** [comment]
giphy

Don't make it hang too long.

File: Controller/Adminhtml/Menu/Delete.php [hunk 1]
```diff
@@ -58,25 +60,11 @@ public function __construct(
...
-            $filterGroupBuilder = $this->filterGroupBuilderFactory->create();
-            $filterGroup = $filterGroupBuilder->addFilter($filter)->create();
-
-            $searchCriteriaBuilder = $this->searchCriteriaBuilderFactory->create();
-            $searchCriteria = $searchCriteriaBuilder->setFilterGroups([$filterGroup])->create();
-
-            $nodes = $this->nodeRepository->getList($searchCriteria);
-            foreach ($nodes->getItems() as $node) {
-                $this->nodeRepository->delete($node);
-            }
```
Comment (TM18) [inline Controller/Adminhtml/Menu/Delete.php:79]: As this block is removed, I think `filterBuilderFactory`, `filterGroupBuilderFactory`, `searchCriteriaBuilderFactory` and `nodeRepository` properties can be removed as well along with `use` statements and constructor assignments.

File: Model/MenuRepository.php [hunk 2]
```diff
@@ -1,66 +1,117 @@
...
     public function __construct(
-        MenuFactory $objectFactory,
-        CollectionFactory $collectionFactory,
-        MenuSearchResultsInterfaceFactory $menuSearchResults
+        MenuFactory $menuFactory,
+        MenuCollectionFactory $menuCollectionFactory,
+        MenuSearchResultsInterfaceFactory $menuSearchResults,
+        ResourceModel\Menu $menuResourceModel = null,
+        NodeCollectionFactory $nodeCollectionFactory = null,
+        ResourceModel\Menu\Node $nodeResourceModel = null
```
Comment (TM18) [inline Model/MenuRepository.php:51]: PHPDoc is missing for `$nodeResourceModel`

**Review by TM18 — CHANGES_REQUESTED** [review]

**Comment by kiber** [comment]
@lbajsarowicz could you please proceed with the requested changes?

**Comment by kiber** [comment]
@lbajsarowicz any updates?

File: Model/MenuRepository.php (hunk 2 above)
Comment (lbajsarowicz) [reply Model/MenuRepository.php:52]: ✔️  Sorted

File: Controller/Adminhtml/Menu/Delete.php (hunk 1 above)
Comment (lbajsarowicz) [reply Controller/Adminhtml/Menu/Delete.php:79]: ✔️  Sorted

**Review by jochhop — APPROVED** [review]

**Comment by rafdol** [comment]
@lbajsarowicz could you merge current develop into your branch, please?

File: Controller/Adminhtml/Menu/Delete.php [hunk 3]
```diff
@@ -1,82 +1,45 @@
...
-        $this->searchCriteriaBuilderFactory = $searchCriteriaBuilderFactory;
     }
 
     /**
      * Dispatch request
      *
-     * @return \Magento\Framework\Controller\ResultInterface|ResponseInterface
-     * @throws \Magento\Framework\Exception\NotFoundException
+     * @return ResultInterface|ResponseInterface
+     * @throws NotFoundException
```
Comment (rafdol) [inline Controller/Adminhtml/Menu/Delete.php:34]: Where is this exception (`NotFoundException`) thrown? I see only `CouldNotDeleteException`, `InvalidArgumentException`, `NoSuchEntityException`.

File: Controller/Adminhtml/Menu/Delete.php [hunk 4]
```diff
@@ -1,82 +1,45 @@
...
      */
     public function execute()
     {
-        $id = $this->getRequest()->getParam('id');
-
         try {
+            $id = $this->getRequestMenuId();
```
Comment (rafdol) [inline Controller/Adminhtml/Menu/Delete.php:39]: The better name for this variable is `$menuId` in my opinion.

**Comment by andrzejborkowski** [comment]
@lbajsarowicz please proceed with the requested changes, then we'll review the code again and merge it to develop branch. Thank you!

**Comment by lbajsarowicz** [comment]
I don't have time to finish this.

**Pull request closed without merging**